glTFLoaderUnitTests/SampleModelsTest.cs

```diff
@@ -51,178 +71,66 @@ private glTFLoader.Schema.Gltf TestLoadFile(string filePath)
+        [TestCase("glTF-pbrSpecularGlossiness")]
+        [TestCase("glTF-techniqueWebGL")]
+        public void SchemaLoad(string subdirectory)
         {
-            foreach (var dir in Directory.EnumerateDirectories(Path.GetFullPath(AbsolutePathToSchemaDir)))
-            {
-                string path = Path.Combine(dir, "glTF-Embedded");
-                if (Directory.Exists(path))
-                {
-                    foreach (var file in Directory.EnumerateFiles(path))
-                    {
-                        TestLoadFile(file);
-                    }
-                }
-            }
-        }
-
-        [Test]
-        public void MaterialsSchemaLoad()
-        {
-            foreach (var dir in Directory.EnumerateDirectories(Path.GetFullPath(AbsolutePathToSchemaDir)))
-            {
-                string path = Path.Combine(dir, "glTF-MaterialsCommon");
-                if (Directory.Exists(path))
-                {
-                    foreach (var file in Directory.EnumerateFiles(path))
-                    {
-                        TestLoadFile(file);
-                    }
-                }
-            }
-        }
-
-        [Test]
-        public void PBRSchemaLoad()
-        {
-            foreach (var dir in Directory.EnumerateDirectories(Path.GetFullPath(AbsolutePathToSchemaDir)))
-            {
-                string path = Path.Combine(dir, "glTF-pbrSpecularGlossiness");
-                if (Directory.Exists(path))
-                {
-                    foreach (var file in Directory.EnumerateFiles(path))
-                    {
-                        TestLoadFile(file);
-                    }
-                }
-            }
-        }
-
-        [Test]
-        public void WebGLSchemaLoad()
-        {
-            foreach (var dir in Directory.EnumerateDirectories(Path.GetFullPath(AbsolutePathToSchemaDir)))
-            {
-                string path = Path.Combine(dir, "glTF-techniqueWebGL");
-                if (Directory.Exists(path))
-                {
-                    foreach (var file in Directory.EnumerateFiles(path))
-                    {
-                        TestLoadFile(file);
-                    }
-                }
-            }
-        }        
-    }
-}
+            foreach (var file in GetTestFiles(subdirectory))
+            {
+                var gltf = TestLoadFile(file);
+
+                // serialization as glTF (in memory)
+                using (var wm = new MemoryStream())
+                {
+                    Interface.SaveModel(gltf, wm);
+
+                    using (var rm = new MemoryStream(wm.ToArray()))
+                    {
+                        gltf = Interface.LoadModel(rm);
+                        Assert.IsNotNull(gltf);
+                    }
+                }
+
+                // serialization as GLB if compatible (in memory)
+                if (gltf.Buffers?.Length == 1 && gltf.Buffers[0].Uri == null)

```

Comment:
I was hoping there would be some support for loading a glTF file and saving it as GLB without additional work. Looking at this unit test, it seems like there is more to be done. Probably a separate PR.

Reply:
Well.. i've been thinking about it, actually. But in order to write as GLB any glTF, it needs to merge all the buffers if there's more than one buffer, and then update all the BufferView[s] to adjust the offsets and indices, then you have to replace the existing buffers with the single one. it means changing the document.

Reply:
Yes, for sure.